File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    int answer = 0;

       public void netWork(boolean[] checked, int[][] computers, int i, int len) {
        for (int j = 0; j < len; j++) {
            if (i != j && computers[i][j] == 1 && !checked[j]) {
                checked[j] = true;
                this.netWork(checked, computers, j, len);
            }
        }
    }

    public int solution(int n, int[][] computers) {
        if (n == 1) {
            return 1;
        }

        int answer = 0;
        boolean[] checked = new boolean[n];

        for (int i = 0; i < n; i++) {
            if (!checked[i]) {
                this.netWork(checked, computers, i, n);
                answer++;
            }
        }
        return answer;
    }

}

File: 프로그래머스/3/43162. 네트워크/네트워크.kt
class Solution {
    lateinit var network : Array<IntArray>
    var size = 0
    var answer = 0
    
    
    fun solution(n: Int, computers: Array<IntArray>): Int {
        network = computers
        size = n

        for (i in 0 until size) {
            if(network[i][i] == 1) {
                this.dfs(i, 0)
                answer += 1
    
            }
        }
        return answer
    }
    
      private fun dfs(y:Int, x:Int){

        if(y < size && x < size ){
            if(network[y][x] == 1){
                network[y][x] = 0
                for(i in 0 until size){
                    this.dfs(x, i)
                }
            } else {
                this.dfs(y, x + 1)
            }
        }
    }

}